modules/exploits/multi/http/rudder_server_sqli_rce.rb

```diff
@@ -0,0 +1,112 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = ExcellentRanking
+  include Msf::Exploit::FileDropper
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Rudder Server SQLI Remote Code Execution',
+        'Description' => %q{
+          This Metasploit module exploits a SQL injection vulnerability in
+          RudderStack's rudder-server, an open source Customer Data Platform (CDP).
+          The vulnerability exists in versions of rudder-server prior to 1.3.0-rc.1.
+          By exploiting this flaw, an attacker can execute arbitrary SQL commands,
+          which may lead to Remote Code Execution (RCE) due to the `rudder` role
+          in PostgresSQL having superuser permissions by default.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Ege Balcı <egebalci@pm.me>' # msf module
+        ],
+        'References' => [
+          ['CVE', '2023-30625'],
+          ['URL', 'https://securitylab.github.com/advisories/GHSL-2022-097_rudder-server/'],
+          ['URL', 'https://nvd.nist.gov/vuln/detail/CVE-2023-30625'],
+        ],
+        'DefaultOptions' => {
+          'SSL' => false,
+          'WfsDelay' => 5
+        },
+        'Platform' => [ 'unix', 'linux' ],
+        'Arch' => [ ARCH_CMD, ARCH_X86, ARCH_X64 ],
+        'Targets' => [
+          ['Automatic', { 'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_netcat' } }],
+          ['Linux', { 'DefaultOptions' => { 'PAYLOAD' => 'cmd/unix/reverse_netcat' } }],
+          ['Windows', { 'DefaultOptions' => { 'PAYLOAD' => 'windows/powershell_reverse_tcp' } }],
+        ],
+        'DefaultTarget' => 0,
+        'Privileged' => false,
+        'DisclosureDate' => '2023-06-16',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [REPEATABLE_SESSION],
+          'SideEffects' => [ARTIFACTS_ON_DISK, IOC_IN_LOGS]
+        }
+      )
+    )
+
+    register_options(
+      [
+        Opt::RPORT(8080),
+        OptString.new('TARGETURI', [true, 'The URI of the Rudder API', '/']),
+        OptString.new('API_USER', [false, 'The Rudder API basic authentication username', 'key']),
+        OptString.new('API_PASS', [false, 'The Rudder API basic authentication password', '']),
+      ]
+    )
+  end
+
+  def check
+    version = get_version
+    if version.match(/^[01]\.[012]/) # v0.*.* - v1.2*
+      Exploit::CheckCode::Appears("Rudder Version: #{version}")
+    else
+      if version == 'Unknown'
+        return Exploit::CheckCode::Unknown
+      end
+
+      Exploit::CheckCode::Safe
+    end
```

Comment:
We prefer the use of `Rex::Version` when comparing version strings in metasploit: 
```suggestion
    return Exploit::CheckCode::Unknown if verion.nil? or version == 'Unknown'
    
    if Rex::Version.new('1.2') >= Rex::Version.new(version.gsub('v', ''))
      Exploit::CheckCode::Appears("Rudder Version: #{version}")
    end
    Exploit::CheckCode::Safe
```